test/src/problems/TransientExternalProblem.C

```diff
@@ -0,0 +1,66 @@
+#include "TransientExternalProblem.h"
+#include "AuxiliarySystem.h"
+
+registerMooseObject("MooseTestApp", TransientExternalProblem);
```

Comment:
I would change the name. (SyncTestExternalProblem?)
The name would be great for pointing to specificities of a transient external problem

but there's no specificity here, you re just using time to get a good test output